master/buildbot/data/builds.py

```diff
@@ -53,8 +80,16 @@ def get(self, resultSpec, kwargs):
             bldr = kwargs['builderid']
             num = kwargs['number']
             dbdict = yield self.master.db.builds.getBuildByNumber(bldr, num)
-        defer.returnValue((yield self.db2data(dbdict))
-                          if dbdict else None)
+        data = yield self.db2data(dbdict) if dbdict else None
+        filters = resultSpec.popProperties()
+        # Avoid to request DB for Build's properties if not specified
+        if filters:
+            try:
+                props = yield self.master.db.builds.getBuildProperties(data['buildid'])
+            except (KeyError, TypeError):
+                props = {}
+            self._injectProperties(data, props, filters)
```

Comment:
Thanks @sa2ajj @tardyp for the precisions,
it is crystal-clear and I perfectly got what you meant now! I totally agree with you ;) 
I'll take care of it.